### PR TITLE
[WIP] Add a script to automatically import integrations from the Beats repo

### PR DESCRIPTION
**Pull request by exekias**
This PR adds a script to import integrations from Metricbeat and Filebeat. Running `import_beats` will walk beats repo, create the proper manifests, and copy over assets for each module.

image

Having the repo checked out in the same root folder as this repo, run to update `dev/package-beats` folder.
```
go run dev/import_beats/ -beatsDir ../beats -publicDir dev/package-beats
```

This is still a work in progress effort with many things TODO:

- [ ] Generate + copy `fields.yml`
- [ ] Come up with a plan for docs
- [ ] Dashboard screenshots
- [ ] Icons
- [ ] Input configs
- [ ] ...and more

Reviews will probably want to focus on `import_beats/main.go` file, as the rest are generated.

### CODE REVIEW

File: dev/import_beats/main.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+		// dashboards
+		srcDashboardPath := filepath.Join(integration.Path, "_meta/kibana/7/dashboard")
+		fmt.Println(srcDashboardPath)
+		srcDashboards, err := ioutil.ReadDir(srcDashboardPath)
+		if err != nil && !os.IsNotExist(err) {
+			log.Fatal(err)
+		}
+
+		if len(srcDashboards) > 0 {
+			dstDashboardPath := filepath.Join(path, "kibana", "dashboard")
```
Comment (ruflin) [inline dev/import_beats/main.go]: We should discuss what is best for you as the desitination dashboard / visualization etc. format. Also should it be encoded or not?

File: magefile.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ var (
 
 	publicDir    = "./public"
 	buildDir     = "./build"
-	packagePaths = []string{"./dev/package-generated/", "./dev/package-examples/"}
+	packagePaths = []string{"./dev/package-generated/", "./dev/package-examples/", "./dev/package-beats/"}
```
Comment (ruflin) [inline magefile.go]: Perfect for now.

File: dev/import_beats/main.go (hunk 1 above)
Comment (exekias) [reply dev/import_beats/main.go]: That's a good question, so far it has been good for development and keeping things in a repo. I would be +1 on keeping them as a pretty json

File: dev/import_beats/main.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+			},
+			Path: path,
+		}
+		i.list[meta.Key] = integration
+	}
+
+	integration.Categories = append(integration.Categories, category)
+
+	// TODO come up with a general description
+	integration.Description = meta.Description
```
[the remark this answers is not in the document]
Comment (exekias) [reply dev/import_beats/main.go]: yes, I think we will need to do this for many things, will try to figure out what's missing

File: dev/import_beats/main.go (hunk 1 above)
Comment (ruflin) [reply dev/import_beats/main.go]: I definitively think we should keep it as pretty json for the development of integrations. What about inside the packages?

Comment (exekias) [reply dev/import_beats/main.go]: I guess as part of the publish process we can do any transformation that is needed. This probably depends on what's best for Kibana?

Comment (ruflin) [reply dev/import_beats/main.go]: I would argue Kibana wants to do as little work as possible so getting already the "right" format would be beneficial form them I would argue.

One thing to take into account here is Kibana will potentially move to the decoded format in the future ...

Comment (ruflin) [reply dev/import_beats/main.go]: My thinking at the moment:

* Decoded in the integrations repo where you build it
* Decoded in the unzippped packages we have here at the moment
* Encoded inside to public dir.

**Comment by ruflin** [comment]
@exekias To move this PR forward, following suggestion. We take a subset of the script you have above that does the following:

* Create the correct directories from the modules for the packages
* Creates a manifest (not necessarly complete)
* Ignore this directory from now from the build process but check in the files

The above should allow us to make many small iterations on the script, like adding datasets, completing manifests etc. and always see the changes it makes. Also not adding it directly to the package build process will mean we can play around with it, without affecting the current packages.

WDYT?

**Comment by exekias** [comment]
Yes! I think we should retake this effort (maybe even from scratch :innocent:). I agree with a plan like that

**Comment by exekias** [comment]
I have opened https://github.com/elastic/package-registry/issues/221 to formalize this effort, feel free to update it. Closing this PR for now, as it still requires some work to be ready

**Pull request closed without merging**